Rakefile

```diff
@@ -227,100 +225,146 @@ namespace :jsondoc do
     mkdir_p gh_pages + "json/stackdriver/master/google/cloud/"
   end
 
-  desc "Copies all gems jsondoc to gh-pages repo in temp dir."
-  task :copy_all, [:tag] => [:init] do |t, args|
-    tag = args[:tag]
-    fail "Missing required parameter 'tag'." if tag.nil?
-    gh_pages = Pathname.new(Dir.home) + "tmp/#{tag}-gh-pages"
+  desc "Copies a gem's jsondoc to gh-pages repo in temp dir."
+  task :copy, [:gem, :version, :gh_pages_dir] do |t, args|
+    gem, version, gh_pages_dir = extract_args args, :gem, :version, :gh_pages_dir
+    gh_pages = gh_pages_path gh_pages_dir
 
-    header "Copying all jsondoc for branch/tag '#{tag}' to '#{gh_pages}'"
-    gems.each do |gem|
-      unless Dir.exist? gh_pages + "json/#{gem}"
-        mkdir_p gh_pages + "json/#{gem}", verbose: true
-      end
-      gem_version = tag.include?("/") ? tag.split("/").last : tag
-      rm_rf gh_pages + "json/#{gem}/#{gem_version}", verbose: true
-      mkdir_p gh_pages + "json/#{gem}/"
-      cp_r "#{gem}/jsondoc", gh_pages + "json/#{gem}/#{gem_version}", verbose: true
+    header "Copying #{gem} jsondoc for '#{gem}/#{version}' to '#{gh_pages}'"
+
+    unless Dir.exist? gh_pages + "json/#{gem}"
+      mkdir_p gh_pages + "json/#{gem}", verbose: true
     end
+    rm_rf gh_pages + "json/#{gem}/#{version}", verbose: true
+    mkdir_p gh_pages + "json/#{gem}/"
+    cp_r "#{gem}/jsondoc", gh_pages + "json/#{gem}/#{version}", verbose: true
+
     cp "docs/manifest.json", gh_pages, verbose: true
     cp "docs/json/home.html", gh_pages + "json", verbose: true
   end
 
-  desc "Assembles google-cloud umbrella package jsondoc, from gems' jsondoc to gh-pages repo in temp dir."
-  task :umbrella, :tag do |t, args|
-    tag = args[:tag]
-    fail "Missing required parameter 'tag'." if tag.nil?
-    gh_pages = Pathname.new(Dir.home) + "tmp/#{tag}-gh-pages"
+  desc "Updates a gem's toc.json with correct tagName."
+  task :toc, [:gem, :version, :gh_pages_dir] => [:copy] do |t, args|
+    gem, version, gh_pages_dir = extract_args args, :gem, :version, :gh_pages_dir
+    gh_pages = gh_pages_path gh_pages_dir
+
+    header "Updating toc.json for '#{gem}/#{version}' in '#{gh_pages}'"
+    toc_path = gh_pages + "json/#{gem}/#{version}" + "toc.json"
```

Comment:
No, `gh_pages` is a `Pathname` instance, and [`Pathname#+`](https://ruby-doc.org/stdlib-2.3.0/libdoc/pathname/rdoc/Pathname.html#method-i-2B) takes care of it.